app/src/store/data.ts

```diff
@@ -159,6 +161,28 @@ export default function useDataStore() {
     lastBlockTimestamp.value = (timestamp as BigNumber).toNumber();
     grants.value = grantsList as Grant[];
     grantRounds.value = grantRoundsList as GrantRound[];
+    const metaPtrs = grants.value.map((grant) => grant.metaPtr);
+    const newMetadata = metaPtrs
+      .filter((metaPtr) => !grantMetadata.value[metaPtr]) // only want new metaPtrs
+      .reduce((prev, cur) => {
+        return {
+          ...prev,
+          [cur]: { status: 'pending' },
+        };
+      }, {});
+
+    grantMetadata.value = { ...grantMetadata.value, ...newMetadata };
+    await Promise.all(
+      Object.keys(newMetadata).map(async (url) => {
+        try {
+          const data = await resolveMetaPtr(url);
+          grantMetadata.value[url] = { status: 'resolved', ...data };
+        } catch (e) {
+          grantMetadata.value[url] = { status: 'error' };
+          console.error(e);
+        }
+      })
+    );
```

Comment:
It feels like lines 174 and 175 should be switched unless I'm misunderstanding what's going on here?

Also one tip that we are using in this file: start by assigning things to intermediate variables such as `_grantMetadata`, then assign to `grantMetadata.value` at the end of the method once all promises have resolved. The reason for that is because the app is reactive to the state, so if you update state as individual promises resolve, the UI will be much jumpier/more frequent updates than if you do it all at once when all promises are resolved

Reply:
Marking this comment resolved, since @wildmolasses explained the reasoning on a call to me: TL;DR is some URLs may resolve slowly or not at all, so it's better to design the UI to not be jumpy and handle this well, as opposed to waiting for everything to resolve at once which may be slow